oracle/oraclestate.go

```diff
@@ -202,8 +203,16 @@ func NewOracleState(cfg *config.Config) *OracleState {
 	}
 }
 
-func (state *OracleState) SaveStateToFile() {
-	path := filepath.Join(StateFolder, StateFileName)
+func (state *OracleState) SaveStateToFile(slot ...uint64) {
```

Comment:
can we have this logic inside already existent function`SaveStateToFile()`? also we avoid having to provide the slot.

in other words SaveStateToFile() shall overide the current state.gob and create a new state_slot_xxx.gob with that state.

actually it would be a copy, so at that point in time state.gob and state_slot_xxx.gob will be identical i tihnk.